### PR TITLE
forth: update generator

### PR DESCRIPTION
**Pull request by leenipper**
For issue #604.

- Update generator:
  Use .Header in template.
  Align template with latest canonical-data.json.

- Update test program:
  Reference testGroups containing the generated test cases.
  Annotate the FAIL output with test group name & test case description.
  Add PASS output for successful cases.

### CODE REVIEW

File: exercises/forth/example_gen.go [hunk 1]
```diff
@@ -22,78 +22,54 @@ func main() {
...
-	Subtraction        section
-	Multiplication     section
-	Division           section
-	CombinedArithmetic section `json:"combined arithmetic"`
-	Dup                section
-	Drop               section
-	Swap               section
-	Over               section
-	UserDefinedWords   section `json:"user-defined words"`
+	Exercise string
```
Comment (petertseng) [inline exercises/forth/example_gen.go]: I wouldn't mind if you omit these fields from the struct if they won't be used in the generator.

Comment (leenipper) [inline exercises/forth/example_gen.go]: I was simply matching for "matching sake", even though unused. Can fix.

**Comment by leenipper** [comment]
I mistakenly removed my repo branch by this name instead of forch pushing it. Sorry about that. Will I need to just do another PR.  I don't see a way to restore this one to usefulness. @petertseng and @robphoenix , thanks for approving before; may need to do it again on a new PR, unless otherwise there is aother way.

**Comment by petertseng** [comment]
Interesting. Well, the changes we approved were at commit c819c2a9d8d612c0846c859e50337e90a766c2c0, I find this out by observing the "View changes" link next to the "approved these changes" lines. If you were to have re-created the `forth-update-generator` at c819c2a9d8d612c0846c859e50337e90a766c2c0 (for example by `git checkout -b forth-update-generator c819c2a9d8d612c0846c859e50337e90a766c2c0` and pushed it to your fork, then you would have been able to reopen this PR.

**Comment by petertseng** [comment]
> `git checkout -b forth-update-generator c819c2a9d8d612c0846c859e50337e90a766c2c0`

Actually I probably mean `git push origin c819c2a9d8d612c0846c859e50337e90a766c2c0:forth-update-generator` because it only needed to be created on your fork; your local branch didn't need to be touched. Then, you can `git push -f origin update-generator` to put 5f3dd6cf2b54b463ee2c8dbdd71dcf11cd675b50 on the fork's  forth-update-generator branch again.

**Comment by leenipper** [comment]
> git push origin c819c2a9d8d612c0846c859e50337e90a766c2c0:forth-update-generator
> git push -f origin update-generator

Will need to first remove same branch on github, then take above steps. Correct?  Local git is still ok, only adjusting remote repo in order to re-open this one.

**Comment by petertseng** [comment]
Right, so at every step of the way the greyed-out "reopen" on this PR button would have indicated the step to take: It had previously said something like "can't reopen because there is another PR using this branch", and next it would have said "can't reopen because the branch diverged since it was deleted", after the push of `c819c2a:forth-update-generator` then reopen would have been possible I think. And now you have successfully put https://github.com/exercism/xgo/pull/617/commits/5f3dd6cf2b54b463ee2c8dbdd71dcf11cd675b50 on this branch, and the only thing left to do was to fix the conflict, guess you did that already by the time I was done typing.

It wasn't necessary to restore this PR of course, but I personally think it's good to equip myself with knowledge. \*shrug\*

**Comment by leenipper** [comment]
Thanks @petertseng for helping me get it straight.  A bit of messy debris left, but I am learning.

**Comment by robphoenix** [comment]
If it's any consolation @leenipper, I get messed up with git all the time. It's one of the things I've learnt most about since contributing to exercism.

**Pull request merged**